crates/ruma-common/src/events/receipt.rs

```diff
@@ -23,6 +23,25 @@ use crate::{receipt::ReceiptType, EventId, MilliSecondsSinceUnixEpoch, UserId};
 #[ruma_event(type = "m.receipt", kind = EphemeralRoom)]
 pub struct ReceiptEventContent(pub BTreeMap<Box<EventId>, Receipts>);
 
+impl ReceiptEventContent {
+    /// Get the receipt for the given user ID with the given receipt type, if it exists.
+    pub fn user_receipt(
+        &self,
+        user_id: &UserId,
+        receipt_type: ReceiptType,
+    ) -> Option<(&EventId, &Receipt)> {
+        for (event_id, receipts) in self.iter() {
+            for (type_, user_receipts) in receipts.iter() {
```

Comment:
You can use `Iterator::find_map` here instead of loops, I think.

Reply:
I thought it would look cleaner as loops, but I can try it like that.

Reply:
It's actually a good thing I took another look at this, there was one unnecessary loop...